### PR TITLE
[Dominion] Set up dominion hurumap-apps and react ui

### PR DESCRIPTION
**Pull request by KhadijaMahanga**
## Description
This PR clean existing dominion hurumap-apps by removing frontend react app, and adds a django react ui for frontend. It also adds the use of mapit on geojson, 

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots
screenshot from 2019-02-25 11-25-10
screenshot from 2019-02-22 11-06-14
screenshot from 2019-02-22 11-06-38

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by phillipahereza** [comment]
@KhadijaMahanga could you move the `dominion_ui` app into the `dominion` app

**Comment by KhadijaMahanga** [comment]
> @KhadijaMahanga could you move the `dominion_ui` app into the `dominion` app

@phillipahereza I have moved dominion_ui into dominon

File: dominion/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  output: {
+      path: path.resolve('./static/bundles/'),
+      filename: "main.js",
+  },
+
+  watch: true,
+
+  plugins: [
+  //  new BundleTracker({filename: './webpack-stats.json'}),
+  //  new HWP({template: path.join(__dirname,'/dominion/templates/homepage.html')})
```
Comment (karimkawambwa) [inline dominion/webpack.config.js]: Minor nitpick, if we don't need it maybe just remove them all together.
Even from `package.json`.

File: dominion/static/js/map_mapit.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                   _.each(features, function(feature) {
+                     self.decorateFeature(feature, area.type, area.country);
+                   });
+
+                   // index by geoid
+                   _.each(features, function(feature) {
+                       featureMap[feature.properties.geoid] = feature;
+                   });
+
+                   if (counter === 0) {
```
Comment (karimkawambwa) [inline dominion/static/js/map_mapit.js:50]: Minor nitpick maybe do the [decrement](https://github.com/CodeForAfrica/HURUmap-apps/pull/135/files#diff-28f11c7b5f6af8ab955cf76000148291R36) and check here so it not in many places when debugging.

File: dominion/static/js/profile_map_mapit.js [hunk 3]
```diff
@@ -1,86 +1,181 @@
-// extend the default Wazimap ProfileMaps object to add mapit support
-
-var BaseProfileMaps = ProfileMaps;
-ProfileMaps = function() {
+var ProfileMaps = function() {
```
Comment (karimkawambwa) [inline dominion/static/js/profile_map_mapit.js:1]: Maybe we should add a story to turn [these](https://github.com/CodeForAfrica/HURUmap-apps/pull/135/files#diff-28f11c7b5f6af8ab955cf76000148291R5) mapit js files into react components.

File: dominion/settings.py [hunk 4]
```diff
@@ -9,16 +10,13 @@
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
-INSTALLED_APPS = ['dominion'] + INSTALLED_APPS
+INSTALLED_APPS = ['dominion', 'dominion.dominion_ui'] + INSTALLED_APPS
```
Comment (karimkawambwa) [inline dominion/settings.py]: Minor nitpick maybe lets rename `dominion_ui` to `frontend`.

File: dominion/dominion_ui/tests.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
```
Comment (karimkawambwa) [inline dominion/dominion_ui/tests.py]: Empty files maybe just don't them it in here until we add logic.

- https://github.com/CodeForAfrica/HURUmap-apps/pull/135/files#diff-0edaf497805b6fd7f0b48727a7bd32bcR6
- https://github.com/CodeForAfrica/HURUmap-apps/pull/135/files#diff-a713d31f2e54514f01aa62cf4f910550R6
- https://github.com/CodeForAfrica/HURUmap-apps/pull/135/files#diff-f88e05aba4295cb63ba614c48f2a35bcR6

**Review by kilemensi — CHANGES_REQUESTED** [review]
Works @KhadijaMahanga, but it doesn't show the main thing we want to see: Multiple React components reacting in a single page. For example, how are we going to add a _React header_ at the top of the page and _React footer_ at bottom of the page while leaving everything in between to Django templates?

Basically, these choices outline in [React recommended Toolchain](https://reactjs.org/docs/create-a-new-react-app.html#recommended-toolchains):

    * If you’re learning React or creating a new single-page app, use Create React App.
    * If you’re building a server-rendered website with Node.js, try Next.js.
    * If you’re building a static content-oriented website, try Gatsby.
    * If you’re building a component library or integrating with an existing codebase, try More Flexible Toolchains.

I think we're on option 4 there.

------------------------------------------------

Other are just minor stuff like we use `yarn` and `yarn.lock` and not `package-lock.json`, `GPLv3`, etc.

**Comment by KhadijaMahanga** [comment]
@karimkawambwa @kilemensi I have made requested changes. I have also updated a screenshot in the description

**Review by phillipahereza — APPROVED** [review]
@KhadijaMahanga LGTM

**Review by karimkawambwa — APPROVED** [review]
🚀

**Pull request closed without merging**